### PR TITLE
[#5][UI] As a user, I can send password recovery email

### LINKED ISSUE #5: [UI] As a user, I can send password recovery email
## Why
The user needs to OAuth2 authenticate before using the application. After the user provides a valid credential (email and password), the authentication service responds with an access token and refresh token. When the given access token expires, the application automatically refreshes the access token with the refresh token.

## Acceptance Criteria
- Implement send recovery email screen.
image
- Email input is required.
- Back button returns to the sign in screen.
- Update the "Forgot?" link in the sign in screen to redirect to send recovery email screen.

### PR DESCRIPTION
**Pull request by hanam1ni**
- Closes #5 

## What happened 👀

- Add reset password page and form
- Add the back navigation component

## Proof Of Work 📹

The following screen record includes:
- Navigate to reset password page
- Form validation and submission
- Back navigate to login page

https://user-images.githubusercontent.com/14077479/163990249-b64d6150-125c-48dd-b699-db4717eaa63c.mov

### CODE REVIEW

File: src/components/FlashNotice/FlashNoticeIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import styles from './FlashNotice.module.css';
+
+const FlashNoticeIcon = ({ type }: { type: FlashType }) => {
+  switch (type) {
+    case 'success':
+      return (
+        <div className={styles['success-icon']}>
+          <FontAwesomeIcon icon={solid('bell')} />
+        </div>
```
[the remark this answers is not in the document]
Comment (hanam1ni) [reply src/components/FlashNotice/FlashNoticeIcon.tsx]: I updated the icon for both success and warning cases to use SVG from Figma.

File: src/hooks/useSession.tsx [hunk 2]
```diff
@@ -15,7 +18,7 @@ const useSession = (): { user: UserProfile | null; loading: boolean } => {
         setloading(false);
       })
       .catch(() => {
-        router.push('/login');
+        redirect && router.push('/login');
       });
   }, []);
```
[the remark this answers is not in the document]
Comment (hanam1ni) [reply src/hooks/useSession.tsx:23]: Updated in [e3a2070](https://github.com/hanam1ni/react-surveyor/pull/54/commits/e3a2070625ce2ae88c526507a391140b514fd783)

File: src/components/FlashNotice/FlashNoticeIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      return (
+        <Image
+          src="/icon/warning.svg"
+          width={24}
+          height={24}
+          alt="warning icon"
+        />
+      );
+  }
+};
```
Comment (malparty) [inline src/components/FlashNotice/FlashNoticeIcon.tsx:22]: What about a separation between the logic and the rendering? :) 

```suggestion
  const iconPath = '', iconAlt = '';
  
  switch (type) {
    case 'success':
      iconAlt = 'success icon';
      iconPath = '/icon/bell.svg';

    case 'warning':
      iconAlt = 'warning icon';
      iconPath = '/icon/warning.svg';
  }
  
  return (
    <Image src={iconPath} width={24} height={24} alt={iconAlt} />
  );
};
```

Moving in that direction, we can also leverage the fully-type advantage from TypeScript:

Somewhere before in the file:
```typescript
  type IconOptions = { iconPath: string; iconAlt: string };

  const iconOptions: Record<FlashType, IconOptions> = {
    success: {
      iconAlt: 'success icon',
      iconPath: '/icon/bell.svg',
    },
    warning: {
      iconAlt: 'warning icon',
      iconPath: '/icon/warning.svg',
    },
  };
```

And then we don't need the switch case (we replaced the switch case by a declarative object):

```typescript  
  const options = iconOptions[type];

  return (
    <Image src={options.iconPath} width={24} height={24} alt={options.iconAlt} />
  );
};
```


The final result is 2 lines longer than the original code, but it has the following advantages:
- retrieving the options can easily be replaced by another system later (e.g. adding a prop to the component, geting advanced options from a hook, etc...)
- code is DRYer 
- adding new types does not impact the UI
- the options of the icon are fully typed — i.e. early warnings and auto-completion :) 


**Note:** As your current code has only 2 types, implementing this suggestion is definitely NOT needed. ;-)
(I just wanted to give it a try — for me to learn also ^^)

**Review by malparty — APPROVED** [review]
Looks good to me! :) 
My last comment is optional~

File: src/hooks/useSession.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanam1ni) [reply src/hooks/useSession.tsx:27]: Added in [5b1310c](https://github.com/hanam1ni/react-surveyor/pull/54/commits/5b1310c320c74d2983d82bfddaf52a1ec321e3b4)

File: src/components/FlashNotice/FlashNoticeIcon.tsx (hunk 3 above)
Comment (hanam1ni) [reply src/components/FlashNotice/FlashNoticeIcon.tsx]: @malparty Nice suggestion. I updated in [6fd0cdf](https://github.com/hanam1ni/react-surveyor/pull/54/commits/6fd0cdfadcce14ceea310c6860469cee70ee7e87), removed switch case and refactored the icon option make the component look much cleaner and organized ✨

Also, TIL about `Record` utility type which very handy. Thank you 👍

Comment (malparty) [reply src/components/FlashNotice/FlashNoticeIcon.tsx]: My pleasure ;-) Happy you like the suggestion :)

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**